File: nodejs/features/log.js

const tracker = require('@middleware.io/node-apm')
tracker.track();

tracker.info('Info sample');
tracker.warn('Warning sample');
tracker.debug('Debugging Sample');
tracker.error('Error Sample');

if (process.env.MW_AUTOGENERATE_LOGGING_DATA) {
    setTimeout(() => {}, 5000);

    for(let i=0; i<300; i++) {    
        tracker.error('Error Loop Sample');
    }
}

tracker.error(new Error('Error sample with stack trace'));

File: golang/features/log/mwlog/log.go
package main

import (
	"fmt"

	"github.com/middleware-labs/golang-apm/logger"
	track "github.com/middleware-labs/golang-apm/tracker"
)

func main() {
	track.Track(
		track.WithConfigTag(track.Service, "your service name"),
		track.WithConfigTag(track.Project, "your project name"),
	)
	fmt.Println("121212")
	logger.Error("Error")
	logger.Info("Info")
	logger.Warn("Warn")
}
